tests/pipelines/test_pipelines_text_to_speech.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2020 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from transformers import (
+    MODEL_FOR_TEXT_TO_SPEECH_MAPPING,
+    TextToSpeechPipeline,
+    pipeline,
+)
+from transformers.testing_utils import (
+    is_pipeline_test,
+    is_torch_available,
+    require_torch,
+    require_torch_or_tf,
+    slow,
+)
+
+from .test_pipelines_common import ANY
+
+
+if is_torch_available():
+    import torch
+
+
+@is_pipeline_test
+@require_torch_or_tf
+class TextToSpeechPipelineTests(unittest.TestCase):
+    model_mapping = MODEL_FOR_TEXT_TO_SPEECH_MAPPING
+
+    @require_torch
+    def test_small_model_pt(self):
+        speech_generator = pipeline(
+            task="text-to-speech", model="microsoft/speecht5_tts", framework="pt", vocoder="microsoft/speecht5_hifigan"
```

Comment:
This is super dubious that we need 2 distinct models to run the pipeline.

I don't know how to fix it, but one model should know about the other or one should be a sound default for all models

Reply:
This is a specificity of `speechT5`, that we shouldn't allow again for new TTS models, and that's why there are some specific cases on my code!